gobblin-hive-registration/src/main/java/gobblin/hive/HiveTable.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.hive;
+
+import java.util.List;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import gobblin.annotation.Alpha;
+import lombok.Getter;
+import lombok.Setter;
+
+
+/**
+ * A class that represents a Hive table.
+ *
+ * @author ziliu
+ */
+@Getter
+@Alpha
+public class HiveTable extends HiveRegistrationUnit {
+
+  @Setter
+  private List<Column> partitionKeys;
+
+  private Optional<String> owner;
+  private Optional<String> tableType;
+  private Optional<Long> retention;
+
+  HiveTable(Builder builder) {
+    super(builder);
+    this.partitionKeys = ImmutableList.<Column> copyOf(builder.partitionKeys);
+    this.owner = Optional.fromNullable(this.props.getProp(HiveConstants.OWNER));
+    this.tableType = Optional.fromNullable(this.props.getProp(HiveConstants.TABLE_TYPE));
+    this.retention = this.props.contains(HiveConstants.RETENTION)
+        ? Optional.of(this.props.getPropAsLong(HiveConstants.RETENTION)) : Optional.<Long> absent();
+  }
+
+  public void setOwner(String owner) {
+    this.owner = Optional.of(owner);
+  }
+
+  public void setTableType(String tableType) {
+    this.tableType = Optional.of(tableType);
+  }
+
+  public void setRetention(long retention) {
+    this.retention = Optional.of(retention);
+  }
+
+  public static class Builder extends HiveRegistrationUnit.Builder<Builder> {
```

Comment:
I see, I ask because I don't want users to get confused between Gobblin's `HiveTable` and Hive's `Table`. Can you add a little more detail into the javadoc about the differences between the two, why Hive's APIs are not used, what the class is intended for, etc.